code/edge/src/androidTest/java/com/adobe/marketing/mobile/ConfigOverridesFunctionalTests.kt

```diff
@@ -0,0 +1,647 @@
+
+        val xdmData = mapOf("testXdmKey" to "testXdmValue")
+
+        val experienceEvent = ExperienceEvent.Builder()
+            .setXdmSchema(xdmData)
+            .setDatastreamConfigOverride(null)
+            .build()
+
+        Edge.sendEvent(experienceEvent, null)
+
+        // verify
+        mockNetworkService.assertAllNetworkRequestExpectations()
+        val resultRequests = mockNetworkService.getNetworkRequestsWith(
+            EXEDGE_INTERACT_URL_STRING,
+            HttpMethod.POST,
+            TIMEOUT_MILLIS
+        )
+        assertEquals(1, resultRequests.size.toLong())
+
+        val payloadJSONData = getPayloadJson(resultRequests[0])
+        val eventJSONData = payloadJSONData.getJSONArray("events").getJSONObject(0)
+        val metaJSONData = payloadJSONData.getJSONObject("meta")
+
+        assertNotNull(metaJSONData.opt("konductorConfig"))
+        // verify that configOverrides is not present in the meta data
+        assertNull(metaJSONData.opt("configOverrides"))
+
+        JSONAsserts.assertExactMatch(expectedXdmDataMap, eventJSONData)
+    }
+
+    @Test
+    @Throws(InterruptedException::class)
+    fun testSendEvent_withXDMDataAndCustomDataAndDatastreamIdOverride_usesDatastreamIdOverride_sendsNetworkRequestWithOriginalSDKConfig() {
+        val responseConnection = mockNetworkService.createMockNetworkResponse(
+            DEFAULT_RESPONSE_STRING,
+            200
+        )
+        mockNetworkService.setMockResponseFor(
+            EXEDGE_INTERACT_URL_STRING,
+            HttpMethod.POST,
+            responseConnection
+        )
+        mockNetworkService.setExpectationForNetworkRequest(
+            EXEDGE_INTERACT_URL_STRING,
+            HttpMethod.POST,
+            1
+        )
+
+        val expectedXdmDataString = """
+            { "xdm" : 
+                {
+                    "testXdmKey": "testXdmValue"
+                }
+            }
+        """.trim()
+
+        val expectedCustomDataString = """
+            { "data" : 
+                {
+                    "testCustomKey": "testCustomValue"
+                }
+            }
+        """.trim()
+
+        val expectedSdkConfigString = """
+            {
+                "sdkConfig" :{
+                    "datastream": {
+                        "original": "1234abcd-abcd-1234-5678-123456abcdef"
+                    } 
+                }
+             }
+        """.trim()
+        val expectedSdkConfigMap = JSONObject(expectedSdkConfigString)
+        val expectedXdmDataMap = JSONObject(expectedXdmDataString)
+        val expectedCustomDataMap = JSONObject(expectedCustomDataString)
+
+        val xdmData = mapOf("testXdmKey" to "testXdmValue")
+        val customData = mapOf("testCustomKey" to "testCustomValue")
+
+        val experienceEvent = ExperienceEvent.Builder()
+            .setXdmSchema(xdmData)
+            .setData(customData)
+            .setDatastreamIdOverride("TestDatastreamId")
```

Comment:
Can you use a real(ish) example here to ensure a string like "5678abcd-abcd-1234-5678-123456abcdef" is set correctly in the URL?